docs/reconciliation.md

```diff
@@ -122,6 +133,18 @@ metadata:
 - **Ordering**: Lower numbers reconcile first: `-2` → `-1` → `0` → `1` → `2`
 - **Dependencies**: Group `N+1` waits for all group `N` resources to be ready
 
+#### Deletion Order
+
+By default, resources are deleted without regard to their readiness groups. To enable ordered deletion, use this annotation:
+
+```yaml
+metadata:
+  annotations:
+    eno.azure.io/ordered-deletion: "true"
+```
+
+With ordered deletion enabled, a resource's deletion is blocked until all resources in **higher** numbered groups have been deleted (the inverse order of reconciliation).
```

Comment:
Do you think there's any scenarios where we'd want something other than reverse readiness-group?